overlord/snapstate/check_snap.go

```diff
@@ -469,18 +469,36 @@ func earlyEpochCheck(info *snap.Info, snapst *SnapState) error {
 	return checkEpochs(nil, info, cur, nil, Flags{}, nil)
 }
 
-func earlyChecks(st *state.State, snapst *SnapState, update *snap.Info, flags Flags) (Flags, error) {
+func earlyChecks(st *state.State, snapst *SnapState, update *snap.Info, comps []snap.ComponentSideInfo, flags Flags) (Flags, error) {
 	flags, err := ensureInstallPreconditions(st, update, flags, snapst)
 	if err != nil {
 		return flags, err
 	}
 
+	if err := ensureSnapAndComponentsAssertionStatus(update.SideInfo, comps); err != nil {
+		return Flags{}, err
+	}
+
 	if err := earlyEpochCheck(update, snapst); err != nil {
 		return flags, err
 	}
 	return flags, nil
 }
 
+func ensureSnapAndComponentsAssertionStatus(si snap.SideInfo, comps []snap.ComponentSideInfo) error {
+	snapAsserted := !si.Revision.Local() && !si.Revision.Unset()
```

Comment:
we usually use whether we have a SnapID to check for this, not the revisions, can we do that here?

Reply:
Yes, although we don’t have an equivalent for components. Is that okay?

Reply:
yes, you could leave a TODO whether we should consider something about components, but right now I'm more worried about being consistent how we check that for snaps